Dashboard/extract.py

```diff
@@ -0,0 +1,119 @@
+"Extract Script: Pulls product information from ASOS webspage"
+import json
+import logging
+import requests
+from bs4 import BeautifulSoup
+
+DEFAULT_TIMEOUT = 30
+
+
+def get_product_page(url: str, headers: dict) -> str | None:
+    """Fetch the HTML content of a product page from a given URL."""
+    if not url:
+        logging.error("URL is empty")
+        return None
+    try:
+        response = requests.get(url, headers=headers, timeout=DEFAULT_TIMEOUT)
+        return response.text
+    except requests.exceptions.RequestException as e:
+        logging.error("A request error has occurred: %s", e)
+    except TimeoutError as e:
+        logging.error("A timeout error has occurred: %s", e)
+    return None
+
+
+def get_soup(url: str, headers: dict) -> BeautifulSoup | None:
```

Comment:
I think it is worth type checking here, because what would happen if these parameters aren't the correct type?